src/main/kotlin/com/openlattice/batching/EntityBatchingService.kt

```diff
@@ -0,0 +1,167 @@
+package com.openlattice.batching
+
+import com.openlattice.data.storage.partitions.PartitionManager
+import com.openlattice.postgres.PostgresColumn.*
+import com.openlattice.postgres.PostgresTable.BATCHES
+import com.openlattice.postgres.PostgresTable.IDS
+import com.openlattice.postgres.ResultSetAdapters
+import com.zaxxer.hikari.HikariDataSource
+import org.slf4j.LoggerFactory
+import java.sql.PreparedStatement
+import java.util.*
+
+const val DEFAULT_BATCHING_SIZE = 8192
+
+/**
+ * Allow performing batch operations.
+ * @author Matthew Tamayo-Rios &lt;matthew@openlattice.com&gt;
+ */
+class EntityBatchingService(
+        val hds: HikariDataSource,
+        partitionManager:PartitionManager) {
+    val partitions = partitionManager.getAllPartitions()
+    companion object {
+        private val logger = LoggerFactory.getLogger(EntityBatchingService::class.java)
+    }
+
+    /**
+     * Adds up to [batchSize] id to the batch of type [batchType] in the [BATCHES] table.
+     *
+     * @param batchType Use group types of batch operations together. An example here might be indexing vs linking indexing
+     * @param idsSql A select sql fragment returns at least [ENTITY_SET_ID],[ID],[LINKING_ID], and [PARTITION] columns for batch entry that needs to be added.
+     * @param batchSize The maximum number of batch entries to retry to read from executing idsSql and insert.
+     * @param bind A bind function that takes a [PreparedStatement] and [Int] offset, performs the required binds for [idsSql] and returns the base after performing the binds.
+     */
+    fun enqueue(
+            batchType: String,
+            idsSql: String,
+            batchSize: Int = DEFAULT_BATCHING_SIZE,
+            bind: (PreparedStatement, Int) -> Int
+    ) {
+        hds.connection.use { connection ->
+            connection.prepareStatement(buildInsertBatch(idsSql)).use { ps ->
+                ps.setString(1, batchType)
+                val offset = bind(ps, 1)
```

Comment:
Does `bind` increment the offset before binding? I feel like I would expect this to be the starting index for bind (ie 2) -- might be with either updating that or updating the comment to explicitly state that this is the last set index, not the starting index